File: tests/rosenbrock/model/rosenbrock.cpp
#include "rosenbrock.h"
#include <stdio.h>
#include <math.h>

double rosenbrock(double *x)
{
  double s = 0.;
  for (int i = 0; i < NDIMS-1; i++)
      s += 100.*pow(x[i+1]-x[i]*x[i], 2) + pow(x[i]-1., 2);

  return s;
}

File: examples/python/test_functions/model/rosenbrock.py
#!/usr/bin/env python
import math

# Minus Rosenbrock
def m_rosenbrock(x):

    result = 0.0
    for i in range(4):
        result += 100.0*pow(x.getParameter(i+1)-x.getParameter(i),2) + pow(1.0 - x.getParameter(i),2)

    x.addResult(-result)
